src/lightning/fabric/strategies/strategy.py

```diff
@@ -399,6 +399,8 @@ def _convert_stateful_objects_in_state(
                 converted = self.get_module_state_dict(module=obj)
             elif isinstance(obj, Optimizer):
                 converted = self.get_optimizer_state(optimizer=obj)
+            elif isinstance(obj, _Stateful):
+                converted = obj.state_dict()
```

Comment:
Note: The load_checkpoint logic already handles `_Stateful`, so the only change needed is here in the saving logic.